server/models/annotation.js

```diff
@@ -70,17 +72,38 @@ annotationSchema.pre('save', function preSave(next) {
   });
 });
 
-annotationSchema.post('save', (annotation, next) => {
+annotationSchema.post('save', function postSave(annotation, next) {
```

Comment:
I redid the flow in this function to chain everything together through .thens. At first I thought it would be OK to call these functions without chaining them to anything if we don't want to worry about catching errors, but I was having a problem where the notification wouldn't get added unless addUserNotification was chained to something, so I decided to chain everything.